src/Shared/Recruit.Vacancies.Client/Infrastructure/Services/ProviderRelationship/ProviderRelationshipsService.cs

```diff
@@ -0,0 +1,132 @@
+        private readonly IJobsVacancyClient _jobsVacancyClient;
+
+        public ProviderRelationshipsService(IOptions<ProviderRelationshipApiConfiguration> configuration,
+            ILogger<ProviderRelationshipApiConfiguration> logger,
+            IJobsVacancyClient jobsVacancyClient)
+        {
+            _configuration = configuration.Value;
+            _logger = logger;
+            _jobsVacancyClient = jobsVacancyClient;
+        }
+
+        public async Task<IEnumerable<EmployerInfo>> GetLegalEntitiesForProviderAsync(long ukprn)
+        {
+            IEnumerable<EmployerInfo> employerInfos = new List<EmployerInfo>();
+            var httpClient = CreateHttpClient(_configuration);
+            var queryData = new { Ukprn = ukprn, Operation = "Recruitment" };            
+            var uri = new Uri(AddQueryString("accountproviderlegalentities", queryData), UriKind.RelativeOrAbsolute);
+
+            try
+            {
+                var response = await httpClient.GetAsync(uri);
+                if (response.IsSuccessStatusCode)
+                {
+                    var content  = await response.Content.ReadAsStringAsync();
+                    var relations = Newtonsoft.Json.JsonConvert.DeserializeObject<ProviderPermissions>(content);
+                    employerInfos =  MapEmployerInfo(relations);
+                    await AppendAddressToLegalEntitiesAsync(employerInfos);
+                }
+
+                _logger.LogError("An invalid response received when trying to get provider relationships");
+            }
+            catch (HttpRequestException ex)
+            {
+                _logger.LogError(ex, "Error trying to retrieve titles.", null);
+            }
+            catch (JsonReaderException ex)
+            {
+                _logger.LogError(ex, $"Couldn't deserialise {nameof(VacancyApiSearchResponse)}.", null);
+            }
+
+            return employerInfos;
+        }
+
+        private async Task AppendAddressToLegalEntitiesAsync(IEnumerable<EmployerInfo> employerInfos)
+        {            
+            foreach(var employerInfo in employerInfos)
+            {
+                var providerLegalEntities = employerInfo.LegalEntities;
+                var employerLegalEntities = await _jobsVacancyClient.GetEmployerLegalEntitiesAsync(employerInfo.Id);
+
+                foreach(var providerLegalEntity in providerLegalEntities)
+                {
+                    var source = employerLegalEntities.FirstOrDefault(e => e.LegalEntityId == providerLegalEntity.LegalEntityId);
+                    if (source != null)
+                    {
+                        providerLegalEntity.Address = source.Address;
+                    }                    
+                }
+            }
+        }
+
+        private IEnumerable<EmployerInfo> MapEmployerInfo(ProviderPermissions relations)
+        {
+            var result = new List<EmployerInfo>();
+
```

Comment:
As the result is not being used, it could be deleted?

Reply:
removed

Reply:
removed